avocado/core/job.py

```diff
@@ -133,6 +133,28 @@ def __init__(self, args=None):
         self._result_events_dispatcher = dispatcher.ResultEventsDispatcher(self.args)
         output.log_plugin_failures(self._result_events_dispatcher.load_failures)
 
+        # Checking whether we will keep the Job tmp_dir or not.
+        basedir = None
+        keep_tmp = False
+        keep_tmp_arg = getattr(self.args, "keep_tmp", None)
+        keep_tmp_setting = settings.get_value('runner.behavior',
+                                              'keep_tmp_files',
+                                              key_type=bool,
+                                              default=False)
+        # '--kepp-tmp on' makes the Job to keep the tmp_dir
+        if keep_tmp_arg == 'on':
+            keep_tmp = True
+        # 'keep_tmp_files = True' in config file makes the Job to keep
+        # the tmp_dir, but only if '--keep-tmp' is not set to 'off'.
+        elif keep_tmp_setting and keep_tmp_arg != 'off':
+            keep_tmp = True
+        # As we are keeping the tmp_dir, let's create it in a stable
+        # location.
+        if keep_tmp:
+            basedir = self.logdir
```

Comment:
> Well first I don't need this to be config option, but if you feel users could benefit from it, I'm fine with keeping it that way.

The config option is already there. This piece of code is trying to accommodate it with the new `--keep-tmp` argument.